src/core/contracts/registry.ts

```diff
@@ -102,6 +107,36 @@ export const getDSNPRegistryUpdateEvents = async (
   });
 };
 
+/**
+ * validates a message against a signature and then checks that the signer has the permissions
+ * specified.
+ * @param signature the signature for the message
+ * @param message the DSNPMessage
+ * @param dsnpId the DSNP Id of the supposed signer
+ * @param permission the permissions to check for
+ * @param blockNumber (optional) the block number at which to perform this check.  Defaults to "latest"
+ */
+export const validateMessage = async (
+  signature: HexString,
+  message: DSNPMessage,
```

Comment:
I think we need a non-DSNPMessage version of this method as well. Could we support it like this:

```suggestion
  message: DSNPMessage | string,
```
...
```typescript
const serialized = typeof message === string ? message : serialize(message);
```

Reply:
This was a deliberate choice; I'd prefer to make a separate generic string message validation because of the sorting.